src/plugins/guided_onboarding/README.md

```diff
@@ -2,9 +2,20 @@
 
 This plugin contains the code for the Guided Onboarding project. Guided onboarding consists of guides for Solutions (Enterprise Search, Observability, Security) that can be completed as a checklist of steps. The guides help users to ingest their data and to navigate to the correct Solutions pages. 
 
-The guided onboarding plugin includes a client-side code for the UI and the server-side code for the internal API. The server-side code is not intended for external use. 
+The guided onboarding plugin includes a client-side code for the UI and the server-side code for the internal API.  
 
-The client-side code registers a button in the Kibana header that controls the guided onboarding panel (checklist) depending on the current state. There is also an API service exposed from the client-side start contract. The API service is intended for external use by other plugins.
+The client-side code registers a button in the Kibana header that controls the guided onboarding panel (checklist) depending on the current state. There is also an API service exposed from the client-side start contract. The API service is intended for external use by other plugins that need to react to the guided onboarding state, for example hide or display UI elements if a guide step is active.
+
+Besides the internal API routes, the server-side code also exposes a function to register guide configs from the server-side setup start contract. This function is intended for external use by any plugin that need to add a new guide or modify an existing one. 
+
+### Adding a new guide
```

Comment:
[nit] What do you think about colocating this near the "Development" section?

Reply:
Yeah, I think it could actually make more sense at the very end of the readme. We can link directly to the section when needed